src/render.ts

```diff
@@ -7,10 +7,17 @@ const canUseNewReactRootApi =
   reactDomVersion &&
   (reactDomVersion.startsWith('18') || reactDomVersion.startsWith('0.0.0'));
 
-export const renderElement = (node, outlet) => {
+declare let __webpack_public_path__: string | undefined;
+
+export const renderElement = async (node: any, outlet: HTMLElement) => {
```

Comment:
The type of `node` can be improved here:
```suggestion
export const renderElement = async (node: React.ReactNode, outlet: HTMLElement) => {
```

Reply:
good catch, changed!

Reply:
@askoufis changed but had to left some casting in as we still support older react versions that use ReactDOM.render and that needs a diff type